### DISCUSSION TITLE
E900v21E多版本和无线问题

### DISCUSSION THREAD

**Discussion by A-nemon-e** [ophub/amlogic-s9xxx-armbian#1639] [Amlogic]
刷了一个e900v21e以后，又从相同渠道购买一个，特地嘱咐要同一批的，没想到还是不一样。两个都是S905L2，但版号为2ATA411的无线是MT7661，1+8G，板号为2ATA410的无线是MT7668。
Issue[#1278](https://github.com/ophub/amlogic-s9xxx-armbian/issues/1278)的老哥也是2ATA410的板子，无线却是螃蟹8822CS。
忘记在哪看到说是mt的驱动有些没有开源（也可能我记错了），我也没linux基础，在armbian的库里找了半天也不知道驱动在哪（丢人）。可否告知哪些无线网卡被支持？是否dtb、安卓u-boot、overload都可能影响无线？
7661的机，刷的早，刷机前（虽然刷成功了只是没wifi）我没仔细看教程更没搞明白启动的原理，但现在里面在跑东西，暂时没机会重新捣鼓，原厂备份也没弄，不过bootloader还是原厂的
7668的机器（这台备份了），看了xcray大佬的帖子，试了各种u-boot和dtb，结果发现原厂的安卓里面干脆就没有支持wifi。。。（那他装着wifi模块干嘛呢？）

> root@p201_iptv:/dev # ip -o link show
> 1: lo: <LOOPBACK,UP,LOWER_UP> mtu 4096 qdisc noqueue state UNKNOWN mode DEFAULT \    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
> 2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT qlen 1000\    link/ether 70:85:40:f2:9e:7d brd ff:ff:ff:ff:ff:ff
> 3: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT \    link/ipip 0.0.0.0 brd 0.0.0.0
> 4: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN mode DEFAULT \    link/sit 0.0.0.0 brd 0.0.0.0
> 5: ip6tnl0@NONE: <NOARP> mtu 1452 qdisc noop state DOWN mode DEFAULT \    link/tunnel6 :: brd ::

难怪换来换去armbian里面用nmcli dev甚至不显示wlan unavailable。。。
这下尴尬了。。刷别的安卓固件不知道会不会好使。。明天试下
新手，请多多指教，谢谢。

**Comment by ophub** [comment]
https://github.com/unifreq/linux-6.1.y/tree/main/drivers/net/wireless

内核支持哪些可以在源码里看到

**Comment by A-nemon-e** [comment]
谢谢。我看到github有别人发安卓的7621驱动，是否可以自己做移植？可行性如何？

**Comment by ophub** [comment]
安卓的没用，找主线内核的。

**Comment by ophub** [comment]
mt7668 wifi和蓝牙测试内核及使用方法： https://github.com/ophub/amlogic-s9xxx-armbian/issues/3610#issuecomment-5091652991
